File: KinematicsYC/Net/Slot.cs
namespace Assets.SingaPort
{
    public class Slot
    {
        public int Bay { get; set; }
        public int Row { get; set; }
        public int Tier { get; set; }

        public Slot(int bay, int row, int tier)
        {
            this.Bay = bay;
            this.Row = row;
            this.Tier = tier;
        }

        public override bool Equals(object? obj)
        {
            if (obj is Slot other)
                return Bay == other.Bay && Row == other.Row && Tier == other.Tier;
            return false;
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(Bay, Row, Tier);
        }

        public override string ToString()
        {
            return $"Slot({Bay},{Row},{Tier})";
        }
    }
}

File: KinematicsYC/Py/slot.py
class Slot:
    """槽位三维坐标 (Bay, Row, Tier)。"""

    __slots__ = ('bay', 'row', 'tier')

    def __init__(self, bay: int, row: int, tier: int):
        self.bay = bay
        self.row = row
        self.tier = tier

    def __repr__(self) -> str:
        return f"Slot(Bay={self.bay}, Row={self.row}, Tier={self.tier})"

    def __eq__(self, other) -> bool:
        if not isinstance(other, Slot):
            return False
        return self.bay == other.bay and self.row == other.row and self.tier == other.tier

    def __hash__(self) -> int:
        return hash((self.bay, self.row, self.tier))
